src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.html

```diff
@@ -0,0 +1,196 @@
+          <tr>
+            <th></th>
+            <th>
+              <label>From</label>
+            </th>
+            <th colspan="2">
+              <label>To</label>
+            </th>
+          </tr>
+
+          <tr ng-repeat="range in vis.params.colorsRange track by $index">
+            <td>
+              <div class="kuiSideBarFormRow__label__colorbox" ng-style="{ 'background-color': getColor($index) }" />
+            </td>
+            <td>
+              <input
+                ng-model="range.from"
+                type="number"
+                class="form-control"
+                name="range.from"
+                greater-than="{{getGreaterThan($index)}}"
+                step="any" />
+            </td>
+            <td>
+              <input
+                ng-model="range.to"
+                type="number"
+                class="form-control"
+                name="range.to"
+                greater-than="range.from"
+                step="any" />
+            </td>
+            <td>
+              <button
+                type="button"
+                ng-click="removeRange($index)"
+                class="btn btn-danger btn-xs">
+                <i class="fa fa-times"></i>
+              </button>
+            </td>
+          </tr>
+        </table>
+
+        <div class="hintbox" ng-show="!vis.params.colorsRange.length">
+          <p>
+            <i class="fa fa-danger text-danger"></i>
+            <strong>Required:</strong> You must specify at least one range.
+          </p>
+        </div>
+
+        <div
+          ng-click="addRange()"
+          class="sidebar-item-button primary">
+          Add Range
+        </div>
+        <div class="text text-center text-info">Note: colors can be changed in the legend</div>
+      </div>
+    </div>
+  </div>
+  <div>
+    <div class="kuiSideBarCollapsibleTitle">
+      <div
+        class="kuiSideBarCollapsibleTitle__label"
+        ng-click="vis.params.valueAxes[0].labels.show = !vis.params.valueAxes[0].labels.show"
+      >
+        <span
+          aria-hidden="true"
+          ng-class="{
+          'fa-caret-down': vis.params.valueAxes[0].labels.show,
+          'fa-caret-right': !vis.params.valueAxes[0].labels.show
+          }"
+          class="fa fa-caret-right kuiSideBarCollapsibleTitle__caret"
+        ></span>
+        <span class="kuiSideBarCollapsibleTitle__text">
+          Show Labels
+        </span>
+      </div>
+      <input aria-label="enable"
+             ng-model="vis.params.valueAxes[0].labels.show"
+             type="checkbox"
+             class="kuiSideBarSectionTitle__action"
+      >
+    </div>
+    <div ng-if="vis.params.valueAxes[0].labels.show" class="kuiSideBarCollapsibleSection">
+      <div class="kuiSideBarSection">
+        <div class="kuiSideBarFormRow">
+          <label class="kuiSideBarFormRow__label" for="rotateLabels">
+            Rotate
```

Comment:
i think its nice as you might have bars too narrow to put anything in horizontally. there was a bug however that labels didn't get hidden when they would no longer fit by height. thats fixed now.